src/Xamarin.Android.Build.Tasks/Tasks/ResolveSdksTask.cs

```diff
@@ -92,12 +96,38 @@ public override bool Execute ()
 				return false;
 			}
 
+			try {
+				Log.LogDebugMessage ($"JavaSdkPath: {JavaSdkPath}");
+				Xamarin.Android.Tools.JdkInfo info = null;
+				try {
+					info = new Xamarin.Android.Tools.JdkInfo (JavaSdkPath);
+				} catch {
+					info = Xamarin.Android.Tools.JdkInfo.GetKnownSystemJdkInfos (this.CreateTaskLogger ()).FirstOrDefault ();
+				}
```

Comment:
@radekdoulik do you know if these calls to find java take a while?

`ResolveSdks` runs on every project, every build. I might have to add some caching logic to this task after we merge this, I think this task takes ~20ms before these changes on my machine.

Reply:
I believe that this should be cached in the `AppDomain`, as we cache other constructs.

Reply:
Yeah @radekdoulik you could do similar to what we have here, using the incoming `JavaSdkPath` as part of the key: https://github.com/xamarin/xamarin-android/blob/9e21b1d07aa6dc7b0caa2cf3ea6124d26e4cfd57/src/Xamarin.Android.Build.Tasks/Tasks/ValidateJavaVersion.cs#L102-L108

I will do this as part of finding the Android SDK, too; in a different PR.